src/fromager/wheels.py

```diff
@@ -311,29 +331,40 @@ def build_wheel(
         build_env=build_env,
         extra_environ=extra_environ,
         req=req,
+        version=version,
         sdist_root_dir=sdist_root_dir,
         build_dir=pbi.build_dir(sdist_root_dir),
-        version=version,
     )
-    # End the timer
+
+    # invalidate uv's cache
+    build_env.clean_cache(req)
+
     wheels = list(ctx.wheels_build.glob("*.whl"))
     if len(wheels) != 1:
         raise FileNotFoundError(
             f"Expected 1 built wheel in {ctx.wheels_build}, got {len(wheels)}"
         )
+    tmp_wheel_file = wheels[0]
 
-    # invalidate uv's cache
-    build_env.clean_cache(req)
+    # validate location and file name
+    if tmp_wheel_file.parent != ctx.wheels_build:
+        raise ValueError(
+            "{wheel_file!} is not in wheels build directory {ctx.wheels_build!r}"
```

Comment:
probably a typo wheelfile should be tmp_wheel_file with file formatting f"{tmp_wheel_file!r} is not in wheels build directory {ctx.wheels_build!r}"